Library/Homebrew/utils.rb

```diff
@@ -121,6 +121,14 @@ def self.git_last_commit
     HOMEBREW_REPOSITORY.cd { `git show -s --format="%cr" HEAD 2>/dev/null`.chuzzle }
   end
 
+  def self.git_last_commit_date
+    HOMEBREW_REPOSITORY.cd { `git show -s --format="%cd" --date=short HEAD 2>/dev/null`.chuzzle }
+  end
+
+  def self.homebrew_version_string
+    "#{HOMEBREW_VERSION} (git revision #{Homebrew.git_head || "unknown"}; last commit #{Homebrew.git_last_commit_date || "unknown"})"
```

Comment:
Let's use a new `git_short_head` method which calls `git rev-parse --short=4 --verify -q HEAD 2>/dev/null` which gives the shortest possible unique SHA for the commit. Also, can we pull the `||` outside of the string (so the line isn't more than 80 characters) and not print two unknowns in the case we don't have a `git_head`; in that case we known the last commit will always be `unknown` so just printing something like `(no git repository)` is a better idea.